pkg/services/sqlstore/migrations/live.go

```diff
@@ -0,0 +1,24 @@
+package migrations
+
+import (
+	. "github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+func addLiveMigrations(mg *Migrator) {
+	liveChannel := Table{
+		Name: "live_channel",
+		Columns: []*Column{
+			{Name: "id", Type: DB_BigInt, Nullable: false, IsPrimaryKey: true, IsAutoIncrement: true},
+			{Name: "org_id", Type: DB_BigInt, Nullable: false},
+			{Name: "channel", Type: DB_NVarchar, Length: 189, Nullable: false},
+			{Name: "data", Type: DB_Text, Nullable: false},
```

Comment:
To avoid adding many migrations in the next few weeks should we include a few columns we are not using yet?

```
{Name: "config", Type: DB_Text, Nullable: false},
{Name: "secure", Type: DB_Text, Nullable: false},
{Name: "type", Type: DB_NVarchar, Length: 16, Nullable: false},
{Name: "last", Type: DB_DateTime, Nullable: false},
```
Config: will be a json blob that holds details on what to do with the channel.  For example, this is where we should save settings like:
* send metrics to grafana cloud
    * loki/cortex?
    * downsample, delay?
* expected push formats, fields, etc
* check thresholds... 
* ...

secure: place to store encrypted keys for config.  For example the API key for hosted grafana

type? (perhaps an attribute of config :thinking: ) we immediatly have three modes:
1. Streaming data frame -- `data` is the schema, and each message will append data
2. Data frame -- each message should replace the whole frame
3. JSON

last:  for stats, it would be nice to see the last time a message was sent on the channel.  


For both the `data` and `last` values -- we likely want to be careful how often they are actually written.  We could delay quite a bit and that is likely fine

Reply:
> send metrics to grafana cloud
> - loki/cortex?
> - downsample, delay?
> 
> secure: place to store encrypted keys for config. For example the API key for hosted grafana

Sorry for lack of context here, but trying to understand what you're aiming at solving. Is this information coming into from the streaming client (that sends data to the channel) or do you configure a channel in Grafana UI? Feels perhaps like mixing a couple different things like streaming to Grafana frontend and send metrics to grafana cloud? 

> To avoid adding many migrations in the next few weeks should we include a few columns we are not using yet?

Sounds like a design doc could be good? Adding columns that you think you would need later could also lead to additional migrations

Reply:
In general -- this is trying to save configuration for how a channel should behave, it is the analog to datasource/plugin config.  

Adding the check for live enabled is a great idea!